### PR TITLE
Add support for repeatInterval to PushNotificationIOS.scheduleLocalNotification (#2)

### PR DESCRIPTION
**Pull request by adamterlson**
Less broken PR.  Sorry about the hassle.  [Original PR](https://github.com/facebook/react-native/pull/2986).

Included update as per @grabbou's and @mkonicek's feedback.

### CODE REVIEW

File: Libraries/PushNotificationIOS/RCTPushNotificationManager.m [hunk 1]
```diff
@@ -40,6 +40,13 @@ + (UILocalNotification *)UILocalNotification:(id)json
   notification.soundName = [RCTConvert NSString:details[@"soundName"]] ?: UILocalNotificationDefaultSoundName;
   notification.userInfo = [RCTConvert NSDictionary:details[@"userInfo"]];
   notification.category = [RCTConvert NSString:details[@"category"]];
+  
+  NSNumber *repeatInterval = details[@"repeatInterval"];
+
+  if (repeatInterval) {
+    notification.repeatInterval = [RCTConvert NSInteger:repeatInterval];
+  }
+
```
Comment (grabbou) [inline Libraries/PushNotificationIOS/RCTPushNotificationManager.m]: This has to be tested, but I am wondering if the following would work:
```
notification.repeatInterval = [RCTConvert NSInteger:details[@"repeatInterval"] ?: <DefaultRepeatIntervalHerE>
```

File: Libraries/PushNotificationIOS/PushNotificationIOS.js [hunk 2]
```diff
@@ -23,6 +23,15 @@ var DEVICE_NOTIF_EVENT = 'remoteNotificationReceived';
...
 var DEVICE_LOCAL_NOTIF_EVENT = 'localNotificationReceived';
 
+const REPEAT_INTERVAL = {
+   year: 4,
+   month: 8,
+   week: 8192,
+   day: 16,
+   hour: 32,
+   minute: 64
+};
```
Comment (grabbou) [inline Libraries/PushNotificationIOS/PushNotificationIOS.js]: What if we just remove that mapping, pass the `string` value directly to Javascript (no need to check the `details` on JS side) and define something like this (note - you have to check proper enum names):
```objc
@implementation RCTConvert (NSCalendarUnit)

RCT_ENUM_CONVERTER(NSCalendarUnit, (@{
  @"era": @(NSCalendarUnitEra),
  @"week": @(NSCalendarUnitWeekday)
}), 0, integerValue)

@end
```

That means we can just:
```objc
notification.repeatInterval = [RCTConvert NSCalendarUnit:details[@"repeatInterval"]];
```
- default value of `0` will be given for null, undefined, and non-matching strings

**Comment by mkonicek** [comment]
Thanks for the review @grabbou!

**Comment by varungupta85** [comment]
Any update on this PR. I am also looking for this feature for my mobile app.

**Comment by facebook-github-bot-9** [comment]
It's been a while since the last commit was reviewed and the labels show this pull request needs review. Based on the blame information for the files in this pull request we identified @nicklockwood as a potential reviewer. Could you take a look please or cc someone with more context?

**Comment by grabbou** [comment]
@adamterlson hey, do you have some time to apply our comments and rebase on top of master? If not - no worries, just let us now so that I can e.g. go ahead and apply neccessary tweaks! It would be really cool to get this shipped asap!

**Comment by adamterlson** [comment]
@grabbou Sorry, I have been disposed recently and would like to find the time for sure.  My apologies for the delay, I hope to make the fix this weekend when I'm back from travel.

**Comment by adamterlson** [comment]
@grabbou Annnddd I didn't have time this weekend.  Sorry, you may need to make these changes on your end if you want them in a timely fashion.

**Comment by jan-froelich** [comment]
Changed the part in RCTPushNotificationManager.m to this: 
`NSNumber *repeatInterval = [RCTConvert NSNumber:details[@"repeatInterval"]];
  if (repeatInterval) {
    notification.repeatInterval = **[RCTConvert NSInteger:repeatInterval];**
  }`

Works great on my device, it would be really great if we could make the tests pass.

**Comment by skleest** [comment]
any updates on this? would love to see this!

**Comment by varun-acintyo** [comment]
@grabbou Could this be considered for the next release?

**Comment by cmcewen** [comment]
this can be closed via #10337

**Comment by jan-happy** [comment]
So will this be in the next release?

**Pull request closed without merging**